contracts/SmartLoan.sol

```diff
@@ -125,6 +126,19 @@ contract SmartLoan is Ownable {
 
   receive() external payable {}
 
+  /**
+    * Sets ERC20Token allowance for the exchange so that it can sell assets
+    * @dev _asset the code of the given asset
+    * @dev _amount amount of the asset to be allowed
+  **/
+  function setExchangeAllowance(bytes32 _asset, uint256 _amount) public onlyOwner{
```

Comment:
Do we need to use allow & transferFrom pattern? Cannot we simply transfer the exact amount to exchange directly (we know the value of the amount) in a similar way we send AVAX for buying assets?

Reply:
Actually we do not and I think it is a good idea to simplify it. I've updated the code accordingly.